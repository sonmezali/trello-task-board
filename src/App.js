import React, {useState, useEffect} from 'react';
import './main.css';
import Board from './components/Board';


const App=()=> {
  const [boards,setBoards] = useState([
         { id: "b1", title:"Board 1" }
        ])
  const [cards,setCards] = useState([])
  const [reRender, setReRender]=useState(false)


//update dragged card's board id when moved to another board
const updateCard=(cardId, newBoardId)=>{   
    cards.find(card => card.id === cardId).boardId = newBoardId;       
    setCards(cards )
    localStorage.setItem('cards', JSON.stringify(cards ))      // each time cards change the board, all data saved in local storage     
    setReRender(!reRender)
    }

const handleAddTaskBoard=()=>{
  const newBoard={id:`b${boards.length+1}`, title:`Board ${boards.length+1}`}
  setBoards([...boards,newBoard])
  localStorage.setItem('boards', JSON.stringify([...boards,newBoard]))
  }

//when the page first renders it calls data from local storage
  useEffect(() => {
 const boardsStorage= localStorage.getItem('boards')
 const cardsStorage= localStorage.getItem('cards')
 boardsStorage&&setBoards(JSON.parse(boardsStorage))
 cardsStorage&&setCards(JSON.parse(cardsStorage))
  },[])      

  return (
    <div className="App">
      <header className="App-header">  
      <h3 className="header-title">Trello</h3>  
      </header>
      <main className="main">
        {boards.map(board =>
            <Board
              key={board.id}
              id={board.id}
              title={board.title} 
              setCards={setCards}
              className='task-board'
              cards={cards}
              filteredCards={cards.filter(c => c.boardId === board.id)}
              handleDrop={updateCard}
            />
        )}        
        <div className="add-board-btn" onClick={handleAddTaskBoard}><span>Add Board</span></div>
      </main>
      
    </div>
  );
}

export default App;
