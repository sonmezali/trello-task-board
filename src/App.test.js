import React from 'react';
import ReactDom,{unmountComponentAtNode} from 'react-dom';
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App'



describe('test App component', () => {
  let container = null;
  beforeEach(() => {  
    container = document.createElement("div");
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  test("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(<App/>, div);
  });

  test('matches snapshot', () => {
      const tree = renderer.create( <App/> ).toJSON();
      expect(tree).toMatchSnapshot();   
  });

  test('renders button correctly', () => {
      const { getByTestId } = render( <App/> )
      expect(getByTestId('add-board')).toHaveTextContent('Add Board')    
  })
})

    