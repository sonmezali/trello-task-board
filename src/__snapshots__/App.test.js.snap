// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test App component matches snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h3
      className="header-title"
    >
      Trello
    </h3>
  </header>
  <main
    className="main"
  >
    <div
      className="task-board"
      id="b1"
      onDragOver={[Function]}
      onDrop={[Function]}
    >
      <div
        className="board-title"
      >
        Board 1
      </div>
      <div
        className="add-card-btn"
        data-testid="add-card"
        onClick={[Function]}
      >
        <span>
          +Add Card
        </span>
      </div>
    </div>
    <div
      className="add-board-btn"
      data-testid="add-board"
      onClick={[Function]}
      type="button"
    >
      <span>
        Add Board
      </span>
    </div>
  </main>
</div>
`;
